### DISCUSSION TITLE
Postmark TLS Configuration Changes

### DISCUSSION THREAD

**Discussion by k7y6t5** [craftcms/postmark#16] [Q&A] [answered]
Just checking in to find out if [Postmark's upcoming TLS configuration changes](https://postmarkapp.com/updates/upcoming-tls-configuration-changes-for-api-users-action-may-be-required) have already been accounted for in this plugin.

**Comment by vladsandu** [comment, marked as answer]
@k7y6t5 behind the scenes, the Postmark Craft plugin uses [wildbit/swiftmailer-postmark](https://github.com/wildbit/swiftmailer-postmark) for sending mail. This library doesn't do any TLS processing and will support all TLS versions as long as the system library supports them.

You need to make sure that your `cURL`/`openSSL` version supports TLS version greater than 1.0. This is the system library responsible for actually connecting via TLS.
This should be supported unless you're using a very old version of `openSSL` - support for TLS 1.1 & 1.2 was added back in 2012 (in OpenSSL 1.0.1).

As mentioned in the [update post](https://postmarkapp.com/updates/upcoming-tls-configuration-changes-for-api-users-action-may-be-required), we’ve set up a **temporary** endpoint at `api-ssl-temp.postmarkapp.com` that has these changes already applied, so you can use it to test/validate your system against it.

### How to test:
The easiest and most implementation-agnostic way to test this is to update your `hosts` file to point at the test endpoint (temporarily), by adding an entry like so:

`18.223.208.249 api.postmarkapp.com`

For any other related questions, please feel free to comment on this thread!
